applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -58,12 +78,58 @@ def AddDofs(self):
         self._add_dynamic_dofs()
         self.print_on_rank_zero("::[ImplicitMechanicalSolver]:: ", "DOF's ADDED")
 
+    def InitializeSolutionStep(self):
+        # Using the base InitializeSolutionStep
+        super(ImplicitMechanicalSolver, self).InitializeSolutionStep()
+
+        # We compute for the different parts (each body can have it own damping)
+        if self.dynamic_settings["determine_rayleigh_damping"].GetBool() and self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] == 1:
```

Comment:
This stuff should go to `Initialize` (you are only executing it once)

Plus this is copy-pasted, please also move to dynamics_solvers_utilities